scripts/dashboard/validate_dashboards_format.py

```diff
@@ -0,0 +1,153 @@
+'''This script will be triggered by a github action outlined in ~/.github/workflows/validate_dashboards_formar.yml.
+It will be used to validate all new json and metadata files pertaining to integration sample dashboards.'''
+
+import json
+import os
+import sys
+import yaml
+
+class MetadataFormattingError(Exception):
+  '''Raised when data in metadata file does not have expected format'''
+  pass
+
+class JsonFormattingError(Exception):
+  '''Raised when data in json file or file name does not have expected format'''
+  pass
+
+class MetadataInJsonMisMatchError(Exception):
+  '''Raised when data in metadata file does not have matching data in a json file'''
+  pass
+
+class JsonInMetadataMisMatchError(Exception):
+  '''Raised when data in json file does not have matching data in a metadata file'''
+  pass
+
+def mapPaths(paths):
+  '''Takes in list of paths and returns dictionary of metadata and dashboard json data, 
+  organized by directories'''
+  map = {}
+
+  for path in paths:
+    pathParts = path.split('/')
+
+    # filter out non-dashboards files
+    if not 'dashboards' == pathParts[1]:
+        continue
+
+    if not pathParts[2] in map.keys():
+      map[pathParts[2]] = {
+        'metadataIds': {},
+        'metadataNames': {},
+        'jsonIds': {},
+        'jsonNames': {},
+        'jsonFiles': []
+      }
+
+    if pathParts[3].split('.')[-1] == "json":
+      map[pathParts[2]]['jsonFiles'].append(pathParts[3])
+
+  return map
+
+def get_sample_dashboards_json(path):
+  '''Checks if file is in json format and returns data in dictionary form'''
+  with open(path) as f:
+    try:
+      dashboards_dict = json.load(f)
+      dashboards_json = json.dumps(dashboards_dict)
+
+      return dashboards_dict
+
+    except JsonFormattingError:
+      print("{} content could not be loaded".format(path))
```

Comment:
I think it would be appropriate to just have the error throw here.

Reply:
done. created refactored function for json validation check and raised the error directly based on return of that refactored check function.